enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/state/snapshot/CoreStateDownloader.java

```diff
@@ -71,97 +76,160 @@ public CoreStateDownloader( LocalDatabase localDatabase, Lifecycle startStopOnSt
     {
+        /* Extract some key properties before shutting it down. */
+        boolean isEmptyStore = localDatabase.isEmpty();
+
+        /*
+         *  There is no reason to try to recover if there are no transaction logs and in fact it is
+         *  also problematic for the initial transaction pull during the snapshot download because the
+         *  kernel will create a transaction log with a header where previous index points to the same
+         *  index as that written down into the metadata store. This is problematic because we have no
+         *  guarantee that there are later transactions and we need at least one transaction in
+         *  the log to figure out the Raft log index (see {@link RecoverConsensusLogIndex}).
+         */
+        if ( commitStateHelper.hasTxLogs( localDatabase.storeDir() ) )
+        {
+            log.info( "Recovering local database" );
+            ensure( localDatabase::start, "start local database" );
+            ensure( localDatabase::stop, "stop local database" );
+        }
+
+        AdvertisedSocketAddress primary;
+        StoreId remoteStoreId;
         try
         {
-            /* Extract some key properties before shutting it down. */
-            boolean isEmptyStore = localDatabase.isEmpty();
-
-            /*
-             *  There is no reason to try to recover if there are no transaction logs and in fact it is
-             *  also problematic for the initial transaction pull during the snapshot download because the
-             *  kernel will create a transaction log with a header where previous index points to the same
-             *  index as that written down into the metadata store. This is problematic because we have no
-             *  guarantee that there are later transactions and we need at least one transaction in
-             *  the log to figure out the Raft log index (see {@link RecoverConsensusLogIndex}).
-             */
-            if ( commitStateHelper.hasTxLogs( localDatabase.storeDir() ) )
-            {
-                log.info( "Recovering local database" );
-                localDatabase.start();
-                localDatabase.stop();
-            }
+            primary = addressProvider.primary();
+            remoteStoreId = remoteStore.getStoreId( primary );
+        }
+        catch ( CatchupAddressResolutionException | StoreIdDownloadFailedException e )
+        {
+            log.warn( "Store copy failed", e );
+            return false;
+        }
 
-            AdvertisedSocketAddress primary = addressProvider.primary();
-            StoreId remoteStoreId = remoteStore.getStoreId( primary );
-            if ( !isEmptyStore && !remoteStoreId.equals( localDatabase.storeId() ) )
-            {
-                throw new StoreCopyFailedException( "StoreId mismatch and not empty" );
-            }
+        if ( !isEmptyStore && !remoteStoreId.equals( localDatabase.storeId() ) )
```

Comment:
We check store id in any request related to catchup. So I do not thing that this check i necessary. Especially now when we use an AddressProvider we comparing against an adress we might not even use later.

Possible we should add store id into to CatchuRequest interface so that we force it on any request that involves catchup

Reply:
However this is not really related to the changes in this PR

Reply:
StoreId isn't included actually on the subsequent CoreSnapshotRequest.